src/Synth/ADnote.cpp

```diff
@@ -927,6 +1104,42 @@ void ADnote::initparameters(WatchManager *wm, const char *prefix)
                 vce.oscposhiFM[k] += oscposhiFM_add;
                 vce.oscposhiFM[k] %= synth.oscilsize;
             }
+
+            // For WaveTable, we use FMSmp as a modulator. This is sometimes
+            // out of [-1,1]. Our modulator *must* be kept in range [0,1]
+            // note that the factor for 1.0 is 2.0 (precomputation)
+            if(NoteVoicePar[nvoice].FMEnabled == FMTYPE::WAVE_MOD)
+            {
+                // how much can we modulate at most if the center of the
+                // modulation is "basefuncpar", while still fitting the
+                // boundaries [0,1.0]?
+                // e.g.: if the basefuncpar is .4, a maximum oscillation
+                // runs through the wavetable from .0 to .8
+                // => max_space would be .4
+                // if the basefuncpar is .9, a maximum oscillations
+                // runs through the wavetable from 0.8 to 1.0
+                // => max_space would be .1
```

Comment:
From a sonic design perspective is there value in letting the user create modulation that goes beyond [0,1] and then clamping it to the range of [0,1]? I suspect you might have already tried out that approach, but this is more of a curiosity than any call for changes.

Reply:
I guess this belongs to the picture your drew below. What I can say is that the current implementation was very useful for me as a user, as I could generate a lot of interesting sounds. I wonder what @friedolino78 thinks. In the end, if we need both, we can also offer different kinds of wave table modulation.

Reply:
That entirely works for me. It looked like a deliberate choice, but I figured I'd verify.